test/replication-luatest/gh_7253_election_long_wal_write_test.lua

```diff
@@ -0,0 +1,155 @@
+local t = require('luatest')
+local server = require('test.luatest_helpers.server')
+local cluster = require('test.luatest_helpers.cluster')
+local fiber = require('fiber')
+
+local wait_timeout = 50
+
+local g = t.group('gh-7253')
+
+local function wait_pair_sync(server1, server2)
+    -- Without retrying it fails sometimes when vclocks are empty and both
+    -- instances are in 'connect' state instead of 'follow'.
+    t.helpers.retrying({timeout = wait_timeout}, function()
+        server1:wait_vclock_of(server2)
+        server2:wait_vclock_of(server1)
+        server1:assert_follows_upstream(server2:instance_id())
+        server2:assert_follows_upstream(server1:instance_id())
+    end)
+end
+
+local function wait_fullmesh(g)
+    wait_pair_sync(g.server1, g.server2)
+end
+
+local function block_next_wal_write_f()
+    box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+end
+
+local function unblock_wal_write_f()
+    box.error.injection.set('ERRINJ_WAL_DELAY', false)
+end
+
+local function server_block_next_wal_write(server)
+    server:exec(block_next_wal_write_f)
+end
+
+local function server_unblock_wal_write(server)
+    server:exec(unblock_wal_write_f)
+end
+
+local function check_wal_is_blocked_f()
+    if not box.error.injection.get('ERRINJ_WAL_DELAY') then
+        error('WAL still is not paused')
+    end
+end
+
+local function server_wait_wal_is_blocked(server)
+    t.helpers.retrying({timeout = wait_timeout}, server.exec, server,
+                       check_wal_is_blocked_f)
+end
+
+g.before_all(function(g)
+    g.cluster = cluster:new({})
+    local box_cfg = {
+        replication_synchro_timeout = 1000,
+        replication_synchro_quorum = 2,
+        replication_timeout = 0.1,
+        replication = {
+            server.build_instance_uri('server1'),
+            server.build_instance_uri('server2'),
+        },
+    }
+    box_cfg.election_mode = 'manual'
+    g.server1 = g.cluster:build_and_add_server({
+        alias = 'server1', box_cfg = box_cfg
+    })
+    box_cfg.election_mode = 'voter'
+    g.server2 = g.cluster:build_and_add_server({
+        alias = 'server2', box_cfg = box_cfg
+    })
+    g.cluster:start()
+
+    g.server1:exec(function()
+        local s = box.schema.create_space('test', {is_sync = true})
+        s:create_index('pk')
+    end)
+    wait_fullmesh(g)
+end)
+
+g.after_all(function(g)
+    g.cluster:drop()
+    g.server1 = nil
+    g.server2 = nil
+end)
+
+g.test_fence_during_confirm_wal_write = function(g)
+    --
+    -- Make the following topology:
+    --   server2 <-> server1 <- server3
+    --
+    g.server2:exec(function()
+        local replication = table.copy(box.cfg.replication)
+        rawset(_G, 'old_replication', table.copy(replication))
+        replication[3] = nil
```

Comment:
With 2 servers replication[3] is nil already, this whole exec call looks redundant.